src/main/java/org/folio/circulation/domain/notice/schedule/FeeFineScheduledNoticeHandler.java

```diff
@@ -100,36 +138,60 @@ private ScheduledNotice getNextRecurringNotice(ScheduledNotice notice) {
     return notice.withNextRunTime(nextRunTime);
   }
 
-  private static class FeeFineRelatedRecords {
+  private static class FeeFineNoticeContext {
+    private final ScheduledNotice notice;
     private Account account;
     private FeeFineAction action;
+    private Loan loan;
```

Comment:
Can these fields be final?

Reply:
No, they are initialized via `.with*` methods.